### DISCUSSION TITLE
Can we have a map with a list value?

### DISCUSSION THREAD

**Discussion by vimalk78** [bpftrace/bpftrace#2471] [Q&A]
I want to keep track of file fds opened by each process. So i use a map with key as `pid` , but how do i keep multiple fds as value?
I suppose i could use a tuple as a value. but then question is how to append to the tuple?

**Comment by fbs** [comment]
The answer here is similar to #2468, value size for hash maps is fixed length so you cannot append keys. 

`@[pid, fd] = 1` might be an option but you'd have to iterate all keys to find out the fds opened by a pid. The benefit is that a `close()` is easily handled by deleting the map key.
Or you might just send events to userspace and handle the aggregation there

**Reply by vimalk78** [reply]
problem with `@[pid, fd] = 1` is that if the process doesnt call `close(fd)` and just exits with `exit_group` only info we have is the `pid` 
thats why i asked about querying map with partial keys in earlier discussion.
